text/0000-at-least-one-feature.md

```diff
@@ -0,0 +1,183 @@
+
+- The old and new features form a partial order
+
+- P_ω takes those partial orders to the partial order of their downsets.
+  (That is, sets of features with the implied features from feature dependencies "filled in", ordered by inclusion.)
+
+- "same names" maps the old downsets to the new downsets, filling in any newly implied features as needed.
+
+- `#[cfg(...)]` is the mapping of feature sets to exposed library interfaces
+
+- "'upcast' library interfaces" forgets whatever unrelated new stuff was added in the new library version
+
+The idea is that going from the old features directly to the old interfaces, or going the "long way" from old features to new features to new interfaces to old interfaces should yield the same result.
+
+For the more part, features can be "interspersed" anywhere the old feature partial order to make the new feature partial order.
+However, this is an exception!
+The old empty downset becomes the new empty downset, which means nothing can be added below it.
+This is the `default-features = false` gotcha!
+
+When we disallow the empty feature set, we are replacing P_ω with the "free join-semilattice" construction.
+We are enriching features with the ∨ binary operator but no ⊥ identity element.
+There is no empty downset becomes empty downset constraint, and thus we are free to add new features below all the others all we want.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+I can't really think of a reason, it's much simpler than the prior attempts!
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+An alternative is not to ban the empty feature set, but ensure it always translates to the empty library.
+I.e. to require that *all* items must be dependent upon *some* feature; everything needs a `cfg`.
+
+Returning to our half-worked-out theory, instead of banning a notion of a ⊥ empty feature set that must be preserved from the old library to the new (removing a requirement), we are adding a *new* requirement that the ⊥ feature set must map to the ⊥ Rust interface.
+We still have the restriction that new features can be added below, but this restriction is no longer a problem:
+there is no point of adding such a new minimal feature because there is nothing left to `cfg`-out!
+
+This solution is more mathematically elegant, but it seems harder to implement.
+It is unclear how Cargo could require the Rust code to obey this property without new infra like the portability lint.
+
+# Prior art
+[prior-art]: #prior-art
+
+This is a well-known problem.
+See just-rejected [#3283](https://github.com/rust-lang/rfcs/pull/3283), 
+and my previous retracted [#3146](https://github.com/rust-lang/rfcs/pull/3146).
+
+I think this is much simpler than the other two.
+
```

Comment:
Wouldn't these go under alternatives? I would expect prior art to include production precedence rather than never-shipped ideas

Reply:
Fair point. Will move.

Reply:
Moved. There is now a large discussion alternatives including the prior RFCs and "negative features".